RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -64,6 +66,168 @@ public OpenSearchClient(ConnectionContext connectionContext) {
         this.failedRequestsLogger = failedRequestsLogger;
     }
 
+    public Version getClusterVersion() {
+        return client.getAsync("/", null)
+            .flatMap(resp -> {
+                try {
+                    return Mono.just(versionFromResponse(resp));
+                } catch (Exception e) {
+                    return Mono.error(new OperationFailed(e.getMessage(), resp));
+                }
+            })
+            .doOnError(e -> log.error(e.getMessage()))
+            .retryWhen(checkIfItemExistsRetryStrategy)
+            .block();
+    }
+
+    public ObjectNode getClusterData() {
```

Comment:
This class is becoming a very large bucket to catch OpenSearch requests, I think making sub-clients could help isolate and organize code like this.

Reply:
Moved this into the RemoteReaderClient to separate out this functionality